LoRa.Message/PHYPayload.cs

```diff
@@ -38,6 +40,134 @@ public PHYPayload(byte[] packet, byte[] nwkSKey, byte[] appSKey, byte[] appKey,
             Mic = new MIC(this);
         }
 
+          byte[] CMAC()
+        {
+            switch (Mhdr.MessageType)
+            {
+                case MessageType.JoinRequest:
+                    throw new NotImplementedException();
+                    break;
+                case MessageType.JoinAccept:
+                    JoinAcceptMessage payload = (JoinAcceptMessage)MacPayload;
+                   
+                    byte[] cmac = new byte[1 +
+                                                           payload.AppNonce.RawData.Length +
+                                                           payload.NetID.RawData.Length +
+                                                           payload.DevAddr.RawData.Length +
+                                                           1 +
+                                                           1
+                                                           ];
+                    Crypto.CryptoService service = new Crypto.CryptoService();
+                    System.Buffer.BlockCopy(Mhdr.RawData.ToArray(), 0, cmac, 0, Mhdr.RawData.Length);
+                    System.Buffer.BlockCopy(payload.AppNonce.RawData.ToArray(), 0, cmac, Mhdr.RawData.Length, payload.AppNonce.RawData.Length);
+                    System.Buffer.BlockCopy(payload.NetID.RawData.ToArray(), 0, cmac, Mhdr.RawData.Length + payload.AppNonce.RawData.Length, payload.NetID.RawData.Length);
+                    System.Buffer.BlockCopy(payload.DevAddr.RawData.ToArray(), 0, cmac, Mhdr.RawData.Length + payload.AppNonce.RawData.Length + payload.NetID.RawData.Length, payload.DevAddr.RawData.Length);
+                    System.Buffer.BlockCopy(payload.DLSettings.RawData.ToArray(), 0, cmac, Mhdr.RawData.Length + payload.AppNonce.RawData.Length + payload.NetID.RawData.Length + payload.DevAddr.RawData.Length, payload.DLSettings.RawData.Length);
+                    System.Buffer.BlockCopy(payload.RxDelay.RawData.ToArray(), 0, cmac, Mhdr.RawData.Length + payload.AppNonce.RawData.Length + payload.NetID.RawData.Length + payload.DevAddr.RawData.Length + payload.DLSettings.RawData.Length, payload.RxDelay.RawData.Length);
+                    return cmac;
+                    break;
+                case MessageType.Data:
+                    throw new NotImplementedException();
+                    break;
+                case MessageType.Proprietary:
+                    throw new NotImplementedException();
+                default:
+                    throw new NotImplementedException();
+            }
+        }
+
+          byte[] SKey()
```

Comment:
This is a candidate for separate class with JoinRequest and JoinAccept messages.